mlflow/gateway/config.py

```diff
@@ -13,9 +14,12 @@
 from mlflow.gateway.base_models import ConfigModel, ResponseModel
 from mlflow.gateway.utils import is_valid_endpoint_name, check_configuration_route_name_collisions
 from mlflow.gateway.constants import MLFLOW_GATEWAY_ROUTE_BASE, MLFLOW_QUERY_SUFFIX
+from packaging import version
 
 _logger = logging.getLogger(__name__)
 
+IS_PYDANTIC_V2 = version.parse(pydantic.__version__) >= version.parse("2.0")
```

Comment:
This doesn't work with pydantic <2.0, right? 
When I run:
```python
import pydantic
from packaging import version

version.parse(pydantic.__version__)
```
in 1.x, pydantic doesn't have a ``_version__`` dunder and raises:

```sh
---------------------------------------------------------------------------
AttributeError                            Traceback (most recent call last)
/var/folders/cd/n8n0rm2x53l_s0xv_j_xklb00000gp/T/ipykernel_12446/2410932362.py in <cell line: 1>()
----> 1 version.parse(pydantic.__version__)

AttributeError: module 'pydantic' has no attribute '__version__'
```

can we use ``pkg_resources.get_distribution("pydantic").version`` instead, since that works for both major releases?

Reply:
It's been two years since https://github.com/pydantic/pydantic/pull/2573 was merged. I think it's fine to assume the `__version__` is avaialble.

Reply:
Right, but only 1 LTS runtime (the latest) supports that version of pydantic (12.2). LTS 11.3 is loaded with pydantic 1.9.2. Won't this fail if AI Gateway is used on those older runtimes?

Reply:
Good point. I think we can use `pydantic.version.VERSION`.

Reply:
I can test the branch after it's updated on some older runtimes just to make sure :)